### PR TITLE
Don't reveal non existent users

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Don't reveal non existent users with a password reset.

**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes, checking [accessibility](https://www.a11yproject.com/checklist/) etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver
- [ ] Add release notes to [unreleased changelog](https://github.com/teamsempo/SempoBlockchain/blob/master/CHANGELOG.md#unreleased)

### CODE REVIEW

**Comment by michielderoos** [comment]
[[Medium - 5.3] Username and Phone Number Enumeration](https://trello.com/c/1UinBkXq/836-medium-53-username-and-phone-number-enumeration)

File: app/server/api/auth_api.py [hunk 1]
```diff
@@ -624,7 +624,7 @@ def post(self):
             if not user:
                 response_object = {
                     'status': 'fail',
-                    'message': 'User not found'
+                    'message': 'invalid password'
                 }
```
Comment (tristanhcole) [inline app/server/api/auth_api.py:628]: This still doesn't fix the issue. See [penetration testing notes](https://drive.google.com/file/d/1e_lD3RzRcii2hbNy_yTwAneSUk40FG6S/view?usp=sharing) and https://github.com/teamsempo/SempoBlockchain/blob/133aaab34f2b4c214d0c57bfb7a1b5abaf1b0426/app/server/api/auth_api.py#L575-L589

**Review by tristanhcole — CHANGES_REQUESTED** [review]

File: app/server/api/auth_api.py (hunk 1 above)
Comment (michielderoos) [reply app/server/api/auth_api.py:628]: D'oh! Good catch @tristanhcole! Added the new error on ln578 too. Think rate limit would be out of scope and for another PR?

Comment (tristanhcole) [reply app/server/api/auth_api.py:628]: Yep, makes sense to have rate limiting in separate PR

**Review by tristanhcole — APPROVED** [review]
LGTM!

File: app/server/api/auth_api.py (hunk 1 above)
Comment (tristanhcole) [reply app/server/api/auth_api.py:628]: Implemented in #703

**Pull request merged**